libraries/rdiff_backup.rb

```diff
@@ -0,0 +1,151 @@
+      attribute :exclude, kind_of: Array, default: []
+      attribute :cron_minute, kind_of: [String, Integer], default: 0
+      attribute :cron_hour, kind_of: [String, Integer], default: 0
+      attribute :cron_day, kind_of: [String, Integer], default: '*'
+      attribute :cron_weekday, kind_of: [String, Integer], default: '*'
+      attribute :cron_month, kind_of: [String, Integer], default: '*'
+    end
+  end
+  class Provider
+    class RdiffBackup < LWRPBase
+      action :create do
+        include_recipe 'yum-epel'
+        %w(rdiff-backup cronolog).each { |p| package p }
+        if new_resource.nrpe
+          include_recipe 'nrpe'
+          cookbook_file ::File.join(node['nrpe']['plugin_dir'],
+                                    'check_rdiff') do
+            mode 0755
+            cookbook 'rdiff-backup'
+            owner node['nrpe']['user']
+            group node['nrpe']['group']
+            source 'nagios/plugins/check_rdiff'
+          end
+          nrpe_check "check_rdiff_job_#{new_resource.name}" do
+            command ::File.join(node['nrpe']['plugin_dir'],
+                                'check_rdiff '
+                               ) + "-w #{new_resource.nrpe_warning} "\
+                                   "-c #{new_resource.nrpe_critical} "\
+                                   "-r #{new_resource.destination} "\
+                                   "-p #{new_resource.nrpe_period} "\
+                                   "-l #{new_resource.nrpe_transferred}"
+          end
+        end
+        secrets = ::Chef::EncryptedDataBagItem.load('rdiff-backup-secrets',
+                                                    'secrets')
+        user new_resource.owner
+        group new_resource.group unless new_resource.owner == new_resource.group
+        sudo new_resource.owner do
+          user new_resource.owner
+          group new_resource.group
+          commands ['/usr/bin/sudo rdiff-backup'\
+                    '--server --restrict-read-only /']
+        end
+        directory new_resource.lock_dir do
+          owner new_resource.owner
+          group new_resource.group || new_resource.owner
+          mode 0755
+          recursive true
+        end
+        
+        directory ::File.join('/home', new_resource.owner, '.ssh') do
+          owner new_resource.owner
+          group new_resource.group || new_resource.owner
+          mode 0700
+        end
+        ssh_user_private_key 'id_rsa' do
+          user new_resource.owner
+          key secrets['ssh-key']
+        end
+
+        directory '/var/log/rdiff-backup' do
+          owner new_resource.owner
+          group new_resource.group
+          mode 0755
+          recursive true
+        end
+
+        [new_resource.destination, ::File.join('/home',
+                                               new_resource.owner,
+                                               'exclude',
+                                               new_resource.fqdn),
```

Comment:
Seems like this bit of code gets replicated a lot (or at least parts of it). Also begs to question if we want to always make ``/home`` the root directory for the homedir for the rdiff user. I think it would make sense to use a local variable for the the homedir at least. I also wonder if we should have an attribute for where the root of the homedir resides.